internal/configuration/tlsmodes.go

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2023 F5 Inc. All rights reserved.
+ * Use of this source code is governed by the Apache License that can be found in the LICENSE file.
+ */
+
+package configuration
+
+type TLSMode int
+
+var TLSModeMap = map[string]TLSMode{
+	"no-tls":  TLSModeNoTLS,
+	"ca-tls":  TLSModeCaTLS,
+	"ca-mtls": TLSModeCaMTLS,
+	"ss-tls":  TLSModeSsTLS,
+	"ss-mtls": TLSModeSsMTLS,
+}
+
+const (
+	TLSModeNoTLS TLSMode = iota
+	TLSModeCaTLS
+	TLSModeCaMTLS
+	TLSModeSsTLS
+	TLSModeSsMTLS
+)
+
+func (t TLSMode) String() string {
+	modes := []string{"no-tls", "ca-tls", "ca-mtls", "ss-tls", "ss-mtls"}
+	if t < TLSModeNoTLS || t > TLSModeSsMTLS {
```

Comment:
line 27 is a map...? 

I'll look at the full code and see what I'm missing.

By-the-by, if we're going to change the weird abbreviated names in the enums, should we expand these names as well? If might force a change to the documentation too...

Reply:
OIC. The trick here is that having the receiver (`(t TLSMode)`) on this makes it the implicit `ToString` for anything of type `TLSMode`. So when you print out the values you're not having to do explicit map look ups.

Reply:
> line 27 is a map...?
> 
> I'll look at the full code and see what I'm missing.
> 
> By-the-by, if we're going to change the weird abbreviated names in the enums, should we expand these names as well? If might force a change to the documentation too...

This is a bad idea, it would lengthen the values that need to be in the config thereby creating a larger surface area for errors.

Reply:
> By-the-by, if we're going to change the weird abbreviated names in the enums, should we expand these names as well? If might force a change to the documentation too...
No I think due to the kebab-casing those were understandable enough.  `ss-tls` made sense in context even though it's not a common abbreviation.

> So when you print out the values you're not having to do explicit map look ups.
Yeah my suggestion is to keep this function but have it operate off of a `map` defined near `TLSModeMap` but going the other direction. While shorter, `t < TLSModeNoTLS || t > TLSModeSsMTLS` depends on the iota-based implementation and makes it so the addition of a new mode would require a change both in `TLSModeMap` as well as down in this method.

This is optional feedback btw.